modules/vault/main.tf

```diff
@@ -80,7 +88,7 @@ resource "aws_kms_key" "kms_vault_unseal" {
             "Effect": "Allow",
             "Principal": {
                 "AWS": [
-                    "${data.aws_iam_user.vault_user.arn}",
+                    "${data.aws_iam_user.vault_user[0].arn}",
```

Comment:
I know this is still a WIP, but probably better to check if `aws_iam_user.vault_user` is a non-empty tuple, before trying to access it, thoughts?

Reply:
Maybe. For good measure. I was coming from that all resources are conditional on the `var.external_vault` condition. So assuming this resource (resource "aws_kms_key" "kms_vault_unseal") gets created all others are getting created as well and due to the dependencies between them, I know it exist.

Reply:
I added the check